### ISSUE TITLE
Remove `import Ember from 'ember';` declarations

### ISSUE THREAD

**Issue by domchristie** [ember-cli/ember-cli#3592]
Having to include `import Ember from 'ember';` in almost every file seems a little unnecessary.

Whilst I agree that being explicit about your dependencies is a good thing, importing Ember seems like a given in an Ember app.

Builds seem to work without them (apart from a jshint error), so could this be a case of adding `Ember` and `Em` to .jshintrc, and remove import statements from blueprints?

**Comment by rwjblue** [comment]
Feel free to add this to your `.jshintrc` file, that will work at this point but we have decided to not make that the default.

In a fairly short timeframe (when Ember 2.0 hits beta in mid June) Ember will be shipping modules based builds and you will not be able to rely on the `Ember` global (because Ember CLI will be using a modules build which does not setup a `window.Ember` global). Having folks import `Ember` in each file helps us make sure Ember CLI apps still function when we make the switch to the modules based build.

**tldr;** Yes, the global works now, but is not the best path forward.

**Comment by domchristie** [comment]
Thanks for your speedy response!

I’m not really up-to-speed with how the module system works, but is it possible to import `Ember` in every module without having to declare it (in v2)?

**Comment by stefanpenner** [comment]
> I’m not really up-to-speed with how the module system works, but is it possible to import Ember in every module without having to declare it (in v2)?

no. Instead `import Ember from 'ember';` it will most likely become something like:

```js
import Component from 'ember/comonent';
import { readOnly } from 'ember/computed';

export default Component.extend({
   ...
});
```

with more granular module imports, we can:

1. shake the module tree of un-used code at build time, this will allow ember and other add-ons to essentially be more pay-as-you-go in-terms of code and assets shipped to the client
2. improved clarity: rather then pulling in a magical global, your code `import`s exactly what it needs.  `import Promise from 'ember/promise'` `import { alias } from 'ember/computed'`.  In addition, the imports should correlate to docs, and the docs when should provide the appropriate import to use said piece of code.
3. static analysis, will further enable tooling to catch issues early.
4. isolation: especially when it comes to 3rd party dependencies, we don't want different libs stomping on each other.
